src/Services/FHIR/FhirProcedureService.php

```diff
@@ -0,0 +1,209 @@
+
+        $encounter = new FHIRReference();
+        $encounter->setReference('Encounter/'.$data['encounter_id']);
+
+        $performedDateTime = new FHIRDateTime();
+        $performedDateTime->setValue($data['date_collected']);
+
+        $recorder = new FHIRReference();
+        $recorder->setReference('Practitioner/'.$data['provider_id']);
+
+        $asserter = new FHIRReference();
+        $asserter->setReference('Practitioner/'.$data['provider_id']);
+
+        $function = new FHIRCodeableConcept();
+        $functionCoding = new FHIRCoding();
+        $functionCoding->setSystem("http://snomed.info/sct");
+        $functionCoding->setCode($data['specialty']);
+        $functionCoding->setDisplay($data['specialty']);
+        $function->addCoding($functionCoding);
+
+        $actor = new FHIRReference();
+        $actor->setReference('Practitioner/'.$data['provider_id']);
+
+        $onBehalfOf = new FHIRReference();
+        $onBehalfOf->setReference('Organization/'.$data['organization']);
+
+        $performer = new FHIRProcedurePerformer();
+        $performer->setActor($actor);
+        $performer->setFunction($function);
+        $performer->setOnBehalfOf($onBehalfOf);
+
+        $reasonCode = new FHIRCodeableConcept();
+        $reasonCodeCoding = new FHIRCoding();
+        $reasonCodeCoding->setSystem("http://snomed.info/sct");
+        $reasonCodeCoding->setCode($data['order_diagnosis']);
+        $reasonCodeCoding->setDisplay($data['order_diagnosis']);
+        $reasonCode->addCoding($reasonCodeCoding);
+
+        $outcome = new FHIRCodeableConcept();
+        $outcomeCoding = new FHIRCoding();
+        $outcomeCoding->setSystem("http://snomed.info/sct");
+        if (strtolower($data['result_status']) == "corrected" || strtolower($data['result_status']) == "final") {
+            $outcomeCoding->setCode("385669000");
+            $outcomeCoding->setDisplay("Successful");
+        } else if (strtolower($data['result_status']) == "incomplete" || strtolower($data['result_status']) == "preliminary") {
+            $outcomeCoding->setCode("385670004");
+            $outcomeCoding->setDisplay("Partially successful");
+        } else {
+            $outcomeCoding->setCode("385671000");
+            $outcomeCoding->setDisplay("Unsuccessful");
+        }
+        $outcomeCoding->setCode($data['body_site']);
+        $outcomeCoding->setDisplay($data['body_site']);
+        $outcome->addCoding($bodySiteCoding);
+
+        $bodySite = new FHIRCodeableConcept();
+        $bodySiteCoding = new FHIRCoding();
+        $bodySiteCoding->setSystem("http://snomed.info/sct");
+        $bodySiteCoding->setCode($data['body_site']);
+        $bodySiteCoding->setDisplay($data['body_site']);
+        $bodySite->addCoding($bodySiteCoding);
+
+        $followUp = new FHIRCodeableConcept();
+        $followUpCoding = new FHIRCoding();
+        $followUpCoding->setSystem("http://snomed.info/sct");
+        $followUpCoding->setCode($data['related_code']);
```

Comment:
what code set is related code?

Reply:
Not quite sure about it. I got this attribute from the below sql: 
![image](https://user-images.githubusercontent.com/36694215/79696634-9cc63a80-829b-11ea-8e08-7cd9abb6a10f.png)

Reply:
Now I remember, jesh!!
These codes originate in compendium(Procedure->Configuration) and is more or less a convenience because i've never seen it used by any compendium import. It will, if populated, be ICD10. Suppose we could call it a follow up because I know of nowhere else to get one.

Reply:
> what code set is related code?

> Not quite sure about it. I got this attribute from the below sql:

Yes, it doesn't seem clear what is intended for that column.  @sjpadgett Do you have example data?

Reply:
This(related code) looks like it meant for billing.. Bottom line, it doesn't belong here. I don't believe we can support follow up. Is it required?

Reply:
> This(related code) looks like it meant for billing.. Bottom line, it doesn't belong here. I don't believe we can support follow up. Is it required?

Can you clarify what you mean by follow up here?

Reply:
the fhir element followUp. I know we can omit the code and use text but, we don't have a follow up to a procedure like make the patient quit eating candy! We do have notes in procedure that we could use for followUp..

Reply:
OK, sure.  The Procedure resource hasn't had a lot of activity lately in the Patient Care WG - and I had forgotten that it even had a followUp element.  Using that element certainly is optional, and I think in OpenEMR I would be inclined to do it in a different way - typically I think it would be in a procedure note (probably not in Procedure.followUp) or the 'P' of a SOAP note.  The definition text on the followUp element says "The follow up may be represented as a simple note or could potentially be more complex, in which case the CarePlan resource can be used."  Whatever the choice, I'm sure it's best to do it as consistently as possible.

Reply:
> We do have notes in procedure that we could use for followUp

There is also a 'note' element in the Fhir Procedure Resource Content to which I have assigned this notes attribute. 

So we could either do away with the followup element or assign the notes attribute to the followUp element if needed.

Reply:
> There is also a 'note' element

This note is a history note from procedure order that could be sent with the order if a test required. Won't do here.
As @rhausam hinted, this note should come from the procedure report(Electronic Reports in our case) and is call patient related notes on sign off report.

Rather than try to explain how these notes are retrieved(kind of tricky) code example is best.
Look at: library\pnotes.inc around L-114